common/meta/types/proto/meta.proto

```diff
@@ -87,10 +83,75 @@ message WatchResponse {
   Event event = 1;
 }
 
+// messages for txn
+message TxnCondition {
+  // compare target enum
+  enum ConditionTarget {
+    Key = 0;
+    Value = 1;
+  }
+
+  // condition result
+  enum ConditionResult {
+    // only used for key compare
+    KeyExist = 0;
+    KeyNotExist = 1;
+
+    // only used for value compare
+    ValueEqual = 2;
+    ValueGreater = 3;
+    ValueLess = 4;
+    ValueNotEqual = 5;    
+  }
```

Comment:
metasrv is a bit different from etcd:
- `KeyExist` can be expressed with `SeqMatch::GE(0)`.
- `KeyNotExist` can be expressed with `SeqMatch::Exact(0)`.

Thus maybe using an expected seq range is more appropriate in our case.

And besides key and value, we need the ability to assert the version(seq) matching an expected version value.

Reply:
Again, my comment is marked as **resolved** as I can not see any changes to this piece of code, and there is no following comment explaining why it is marked **resolved**.

Reply:
i think change `KeyExist` to a seq compare is not intuition for user, but fine, i refactor the Condition for just Value or Seq compare.

Reply:
It will always be OK to accept a suggestion or not, by updating the code or explaining why it does not need to be solved.

But in this case, the most important thing is the ability to assert the condition of a `seq`, which is not negligible.

Reply:
`seq` is a type of uint64

Reply:
> `seq` is a type of uint64

Right, so?